src/librustc_middle/ty/context.rs

```diff
@@ -1094,23 +1095,12 @@ impl<'tcx> TyCtxt<'tcx> {
         let mut providers = IndexVec::from_elem_n(extern_providers, max_cnum + 1);
         providers[LOCAL_CRATE] = local_providers;
 
-        let def_path_hash_to_def_id = if s.opts.build_dep_graph() {
-            let def_path_tables = crates
-                .iter()
-                .map(|&cnum| (cnum, cstore.def_path_table(cnum)))
-                .chain(iter::once((LOCAL_CRATE, definitions.def_path_table())));
-
-            // Precompute the capacity of the hashmap so we don't have to
-            // re-allocate when populating it.
-            let capacity = def_path_tables.clone().map(|(_, t)| t.size()).sum::<usize>();
-
-            let mut map: FxHashMap<_, _> =
-                FxHashMap::with_capacity_and_hasher(capacity, ::std::default::Default::default());
-
-            for (cnum, def_path_table) in def_path_tables {
-                def_path_table.add_def_path_hashes_to(cnum, &mut map);
-            }
-
+        let local_def_path_hash_to_def_id = if s.opts.build_dep_graph() {
+            let mut map = FxHashMap::with_capacity_and_hasher(
+                definitions.def_path_table().size(),
+                ::std::default::Default::default(),
```

Comment:
```suggestion
                Default::default(),
```
(Not sure why the fully qualified version was used.)